### ISSUE TITLE
Ajax update order review doesn't reload the page

### ISSUE THREAD

**Issue by eliasjnior** [woocommerce/woocommerce#23888]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Add in any theme function or custom plugin the code to force refresh cart:

```php
add_action('wp', 'my_custom_action')';

function my_custom_action() {
    WC()->session->set( 'reload_checkout' );
}
```
2. Add some product and go to checkout.
3. Page won't be reloaded.

## Expected/actual behavior

Checkout should get a infinity loop, because always the response from `POST /?wc-ajax=update_order_review` should be something like this:

```json
{
   "result": "success",
   "message": "",
   "reload": "true",
   "fragments": {}
}
```

The problem is that reload have a **string** value of "true", not a **boolean** value. But in `checkout.js` we have a strict check:

```
if ( data && true === data.reload ) {
   window.location.reload();
   return;
}
```

The fix is easy, on `class-wc-ajax.php:381` we should send the response as a boolean, not a string:

```php
$reload_checkout ? "true" : "false"
$reload_checkout ? true : false
```

## Isolating the problem

- [x] This bug happens with only WooCommerce plugin active
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>

### WordPress Environment ###

WordPress address (URL): http://paypal.localhost
Site address (URL): http://paypal.localhost
WC Version: 3.6.4
Log Directory Writable: ✔
WP Version: 5.2.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: pt_BR
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.25 (Debian)
PHP Version: 7.3.6
PHP Post Max Size: 50 MB
PHP Time Limit: 0
PHP Max Input Vars: 2000
cURL Version: 7.52.1
OpenSSL/1.0.2r

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.15-MariaDB-1:10.3.15+maria~bionic
Max Upload Size: 50 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ O seu servidor não possui a classe SoapClient ativada - alguns métodos de pagamento que usam SOAP podem não funcionar como esperado.
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.6.4
WC Database Prefix: wp_
MaxMind GeoIP Database: ✔
Tamanho total do banco de dados: 4.81MB
Tamanho dos dados do banco de dados: 3.28MB
Tamanho do índice do banco de dado: 1.53MB
wp_woocommerce_sessions: Dados: 0.06MB + Índices: 0.02MB
wp_woocommerce_api_keys: Dados: 0.02MB + Índices: 0.03MB
wp_woocommerce_attribute_taxonomies: Dados: 0.02MB + Índices: 0.02MB
wp_woocommerce_downloadable_product_permissions: Dados: 0.02MB + Índices: 0.06MB
wp_woocommerce_order_items: Dados: 0.02MB + Índices: 0.02MB
wp_woocommerce_order_itemmeta: Dados: 0.08MB + Índices: 0.09MB
wp_woocommerce_tax_rates: Dados: 0.02MB + Índices: 0.06MB
wp_woocommerce_tax_rate_locations: Dados: 0.02MB + Índices: 0.03MB
wp_woocommerce_shipping_zones: Dados: 0.02MB + Índices: 0.00MB
wp_woocommerce_shipping_zone_locations: Dados: 0.02MB + Índices: 0.03MB
wp_woocommerce_shipping_zone_methods: Dados: 0.02MB + Índices: 0.00MB
wp_woocommerce_payment_tokens: Dados: 0.02MB + Índices: 0.02MB
wp_woocommerce_payment_tokenmeta: Dados: 0.02MB + Índices: 0.03MB
wp_woocommerce_log: Dados: 0.02MB + Índices: 0.02MB
wp_commentmeta: Dados: 0.02MB + Índices: 0.03MB
wp_comments: Dados: 0.05MB + Índices: 0.09MB
wp_links: Dados: 0.02MB + Índices: 0.02MB
wp_options: Dados: 2.08MB + Índices: 0.05MB
wp_postmeta: Dados: 0.33MB + Índices: 0.30MB
wp_posts: Dados: 0.08MB + Índices: 0.06MB
wp_termmeta: Dados: 0.02MB + Índices: 0.03MB
wp_terms: Dados: 0.02MB + Índices: 0.03MB
wp_term_relationships: Dados: 0.02MB + Índices: 0.02MB
wp_term_taxonomy: Dados: 0.02MB + Índices: 0.03MB
wp_usermeta: Dados: 0.02MB + Índices: 0.03MB
wp_users: Dados: 0.02MB + Índices: 0.05MB
wp_wc_admin_notes: Dados: 0.02MB + Índices: 0.00MB
wp_wc_admin_note_actions: Dados: 0.02MB + Índices: 0.02MB
wp_wc_customer_lookup: Dados: 0.02MB + Índices: 0.03MB
wp_wc_download_log: Dados: 0.02MB + Índices: 0.03MB
wp_wc_order_coupon_lookup: Dados: 0.02MB + Índices: 0.03MB
wp_wc_order_product_lookup: Dados: 0.02MB + Índices: 0.06MB
wp_wc_order_stats: Dados: 0.02MB + Índices: 0.05MB
wp_wc_order_tax_lookup: Dados: 0.02MB + Índices: 0.03MB
wp_wc_product_meta_lookup: Dados: 0.02MB + Índices: 0.09MB
wp_wc_webhooks: Dados: 0.02MB + Índices: 0.02MB

### Post Type Counts ###

attachment: 20
customize_changeset: 1
nav_menu_item: 4
page: 9
post: 3
product: 12
revision: 9
scheduled-action: 16
shop_coupon: 1
shop_order: 61
wpcf7_contact_form: 1

### Security ###

Secure connection (HTTPS): ❌
					Sua loja não está usando HTTPS. Saiba mais sobre HTTPS e certificados SSL.
Hide errors from visitors: ✔

### Active Plugins (4) ###

Portal de Pagamentos PayPal: por PayPal – 1.0.0 – Não testado com a versão ativa do WooCommerce
WooCommerce Extra Checkout Fields for Brazil: por Claudio Sanches – 3.6.1 – Não testado com a versão ativa do WooCommerce
Portal do Stripe para WooCommerce: por WooCommerce – 4.2.0
WooCommerce: por Automattic – 3.6.4

### Inactive Plugins (2) ###

Checkout Transparente do PayPal: por PayPal – 1.6.5
WooCommerce Admin: por WooCommerce – 0.12.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: BRL (R$)
Currency Position: left
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Base da loja: #43 - /loja/
Carrinho: #44 - /carrinho/
Finalizar compra: #45 - /finalizar-compra/
Minha conta: #46 - /minha-conta/
Termos e condições: ❌ Página não definida

### Theme ###

Name: Storefront
Version: 2.4.6 – 2.5.0 está disponível
Author URL: https://woocommerce.com/
Child Theme: ❌ – Recomendamos você utilizar um tema filho no caso de você estiver personalizando o WooCommerce em um tema ascendente. Veja como criar um tema filho
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 14
Oldest: 2019-05-20 18:54:11 +0000
Newest: 2019-05-20 18:56:06 +0000

Pending: 1
Oldest: 2019-06-05 17:36:35 +0000
Newest: 2019-06-05 17:36:35 +0000

Canceled: 1
Oldest: 2019-06-05 17:33:24 +0000
Newest: 2019-06-05 17:33:24 +0000

In-progress: 0
Oldest: –
Newest: –

Failed: 0
Oldest: –
Newest: –

</details>

## Online Testing

I setup a envinronment in this URL: https://paypal.eliasjr.dev/

You can add some item to cart and go to checkout, open NETWORK tab and see the request.

The plugin installed there is this: https://gist.github.com/eliasjnior/d8891e4e9a0a3b60d98ebe5cebfd7206

**Comment by kloon** [comment]
As of WC 3.6 the session class is only loaded when the cart is loaded. If you call it on pages the cart is not required ie every page not checkout or cart it will not load.

Try calling `wc_load_cart()` before calling `WC()->session->set( 'reload_checkout' )` that should make the cart and session classes load.

**Comment by eliasjnior** [comment]
@kloon sorry, I think you understand that wrong.

I'm working on an official integration for PayPal Brazil and we found that problem. The problem *is not* with session, the problem is the way this variable is sent to frontend.

Take a look on `includes/class-wc-ajax.php:377+`:

```
wp_send_json(
    array(
        'result' => empty($messages) ? 'success' : 'failure',
        'messages' => $messages,
        'reload' => $reload_checkout ? 'true' : 'false',
        'fragments' => apply_filters(
            'woocommerce_update_order_review_fragments',
            array(
                '.woocommerce-checkout-review-order-table' => $woocommerce_order_review,
                '.woocommerce-checkout-payment' => $woocommerce_checkout_payment,
            )
        ),
    )
);
```

You *always* send the boolean value for `reload` as a *string*, not as a boolean. But in JS you strict check for boolean!

```
if ( data && true === data.reload ) {
   window.location.reload();
   return;
}
```

**Comment by kloon** [comment]
Ah okay, I see now. Keeping the issue closed but will tag it as having a pull request and being a bug.
